recipes-apps/ovmenu-ng/files/openvario-57-lvds/ovmenu-ng.sh

```diff
@@ -318,6 +321,14 @@ function download_igc_files() {
 	dialog --backtitle "OpenVario" --title "Result" --tailbox /tmp/tail.$$ 30 50
 }
 
+# Delete only *.igc files
+function delete_igc_files() {
+	echo "Deleting IGC files ..." > /tmp/tail.$$
+	/usr/bin/delete-igc.sh >> /tmp/tail.$$ &
+	dialog --backtitle "OpenVario" --title "Result" --tailbox /tmp/tail.$$ 30 50
```

Comment:
I think destructive operation like this needs at least a confirmation dialog. You don't want to accidentally delete your files if you accidentally miss Download_IGC action.

Reply:
Hi kedder and thanks!
I'll see how to add a confirmation dialog just in case.

Reply:
At least in a club environment easily accessible destructive operations could be an issue. Do we really need a "Delete IGC" option in ovmenu? Disk space on the µSD card is no issue so this is probably more about not having to download all igc files again and again with "Download IGC"? 
What about a "Download newest IGC" option instead? File dates are incorrect on the OV (no battery buffered RTC) but the date is in the IGC file name. Or maybe someday we will be able to select files from a list and upload / download them with a button press.

Reply:
I'm kind of with @mihu-ov on this. In what circumstances would you want to delete all your flight logs?

Reply:
Can we merge the "Done" message even if we have no final decision on the "Delete IGC" option? I would consider that a small but very nice improvement.

Reply:
Hi guys. I've installed two OV in our club´s sailplanes and the question about how to delete IGC files was one of the most often asked questions I had. I agree a destructive method should have a confirmation. I also believe the option to delete must be there. 
All the XCSoar logs are not IGC so they are useless for contests. When you try to download a log from a FLARM to openvario (because FLARM is not accessible to download directly) you have to wait a full IGC download to get the file you want. The more the plane flies, the more you have to wait. I've added the option and what most of our guys do is delete all IGC, fly and then download the FLARM log to send to contest scoring. You are not deleting the contest valid FLARM or logger IGC file so in the worst case scenario no harm is done.
I also agree that a kinda "File Commander" to select which files to download or to delete would be the best option. I just saw this as a temporary patch for the time being.
Sorry if I posted this as a pull request. It's my first time using GitHub as part of a group and I did not know how to share these ideas. If you prefer redmine or XCSoar forum to share please let me know.

Thanks for your time and comments!!
BR

PD: I can create a Branch with only "DONE!" changes so we can merge.

Reply:
> When trying to download a log from FLARM to openvario (because FLARM is not fisically accessible for download) you have to wait a full IGC download to get the file you want. The more the plane flies, the more you have to wait.

@iglesiasmarianod, but isn't the root of the problem is that OpenVario downloads *all* the logs all the time, even though most of the time you need only the last one (or all logs for the last day, to generalize). If we would have this option, nobody would want to delete all files, right?

Reply:
What if we don´t delete but move the old logs from ~/.xcsoar/logs to some other directory as a compromise? We had several club members who downloaded their flight from the logger to the OV but not from OV to USB drive and realized it only a few days later. Their old logs could still be copied to USB in a shell and not be lost.

Reply:
@kedder you are right! I couldn't figure out yet how to "know" what the last file is without RTC.
I started wondering about filling a Menu dialog with filenames inside the log folder.
Something like this (found in stackoverflow):

filter() 
{
    NUM=0
    ls | egrep 'somefilter' | while read i ; do
        echo $'\t'$NUM$'\t'$i
        NUM=$(($NUM+1))
    done
}

LINES=$(filter)

IFS=$'\n\t'

dialog --backtitle "Linux Shell Script" --title "Main\
Menu" --menu "Move using [UP] [DOWN],[Enter] to\
Select" 15 50 3 $LINES 2>/tmp/menuitem.$$

unset IFS

And to retrieve the filename to download this

ITEM=$(cat /tmp/menuitem.$$)
FILE=$(echo "$LINES" | sed -n 's/\t'$ITEM'\t//p')

Do you think it might worth a try?

Reply:
@iglesiasmarianod, this would make sense to me. This would be very similar to how you download logs from Flarm - just pick files you want to download. 

Just make sure newest files come first in that list (sort by filename in reverse order).